core/src/main/java/io/grpc/util/AdvancedTlsX509KeyManager.java

```diff
@@ -126,10 +124,15 @@ public void updateIdentityCredentials(PrivateKey key, X509Certificate[] certs)
    * @return an object that caller should close when the file refreshes are not needed
    */
   public Closeable updateIdentityCredentialsFromFile(File keyFile, File certFile,
-      long period, TimeUnit unit, ScheduledExecutorService executor) {
+      long period, TimeUnit unit, ScheduledExecutorService executor) throws IOException,
+      GeneralSecurityException {
+    UpdateResult newResult = readAndUpdate(keyFile, certFile, 0, 0);
+    if (!newResult.success) {
+      throw new GeneralSecurityException("The initial update was not successful");
```

Comment:
As a user, it isn't clear what to do with this error. As we discussed, `success` really has more of `updated` semantics. Maybe use the language, "Files were unmodified before their initial update. Probably a bug."? Ditto for trust manager.